strawberryfields/apps/similarity.py

```diff
@@ -412,119 +527,218 @@ def prob_event_mc(
     if not 0 <= loss <= 1:
         raise ValueError("Loss parameter must take a value between zero and one")
+
+    if mc_samples:
+        return [
+            _prob_event_mc(graph, photon_number, max_count_per_mode, n_mean, mc_samples, loss)
+            for photon_number in event_photon_numbers
+        ]
+
+    return [
+        _prob_event_exact(graph, photon_number, max_count_per_mode, n_mean, loss)
+        for photon_number in event_photon_numbers
+    ]
+
+
+def feature_vector_orbits_sampling(samples: list, list_of_orbits: list) -> list:
+    r"""Calculates feature vector made of given orbits with respect to input samples.
+
+    The feature vector is composed of orbit probabilities reconstructed by measuring the
+    occurrence of given orbits in the input ``samples``.
+
+    **Example usage:**
+
+    >>> from strawberryfields.apps import data
+    >>> samples = data.Mutag0()
+    >>> feature_vector_orbits_sampling(samples, [ [1,1], [2], [2,1,1], [1,1,1,1] ])
+    [0.19035, 0.0, 0.05175, 0.1352]
+
+    Args:
+        samples (list[list[int]]): a list of samples
+        list_of_orbits (list[list[int]]): a list of orbits
+
+    Returns:
+        list[float]: a feature vector made up of estimated orbit probabilities
+    """
+    if len(list_of_orbits) <= 0:
+        raise ValueError("List of orbits must have at least one orbit")
+    if all(isinstance(elem, list) for elem in list_of_orbits) == False:
+        raise ValueError("List of orbits should only consist of orbits")
+    if any(min(item) < 0 for item in list_of_orbits):
+        raise ValueError("Cannot request orbits with photon number below zero")
+
+    n_samples = len(samples)
+
+    sample_orbits = [sample_to_orbit(each) for each in samples]
+    count_orbits = Counter(map(tuple, sample_orbits))
+
+    return [count_orbits[tuple(orb)] / n_samples for orb in list_of_orbits]
+
+
+def feature_vector_events_sampling(
+    samples: list, event_photon_numbers: list, max_count_per_mode: int = 2
+) -> list:
+    r"""Calculates feature vector made of given events with respect to input samples.
+
+    The feature vector is composed of event probabilities reconstructed by measuring the
+    occurrence of given events in the input ``samples``.
+
+    **Example usage:**
+
+    >>> from strawberryfields.apps import data
+    >>> samples = data.Mutag0()
+    >>> feature_vector_events_sampling(samples, [2, 4, 6], 2)
+    [0.19035, 0.2047, 0.1539]
+
+    Args:
+        samples (list[list[int]]): a list of samples
+        event_photon_numbers (list[int]): a list of events described by their total photon number
+        max_count_per_mode (int): maximum number of photons per mode for all events
+
+    Returns:
+        list[float]: a feature vector made up of estimated event probabilities in the
+        same order as ``event_photon_numbers``
+    """
+    if type(event_photon_numbers) != list:
+        raise ValueError("Event photon numbers must be given as a list of integers")

```

Comment:
This check is too strong, recommend to remove.

Also, you can use `isinstance(event_photon_numbers, list)`